core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1906,7 +1906,9 @@ private long loadMetadataIfNotExistAndJournal(LockedInodePath inodePath,
     boolean loadDirectChildren = false;
     if (inodeExists) {
       try {
-        loadDirectChildren = inodePath.getInode().isDirectory() && options.isLoadDirectChildren();
+        Inode inode = inodePath.getInode();
+        loadDirectChildren =
+            inode.isDirectory() && inode.isPersisted() && options.isLoadDirectChildren();
```

Comment:
1. could you also update the javadoc
2. this function is so important and seems tricky, could you add a unit test for this? (not necessary in this PR)